frontend/packages/rhoas-plugin/src/topology/rhoas-data-transformer.ts

```diff
@@ -0,0 +1,115 @@
+import { EdgeModel, Model, NodeModel } from '@patternfly/react-topology';
+import { apiVersionForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { getTopologyNodeItem } from '@console/topology/src/data-transforms/transform-utils';
+import { OverviewItem } from '@console/shared/src';
+import { TopologyDataObject, TopologyDataResources } from '@console/topology/src/topology-types';
+import {
+  KAFKA_WIDTH,
+  KAFKA_HEIGHT,
+  KAFKA_PADDING,
+  TYPE_MANAGED_KAFKA_CONNECTION,
+} from './components/const';
+import { KafkaConnectionModel } from '../models';
+import { edgesFromServiceBinding } from '@console/topology/src/operators/operators-data-transformer';
+import { TYPE_SERVICE_BINDING } from '@console/topology/src/const';
+
+const KAFKA_PROPS = {
+  width: KAFKA_WIDTH,
+  height: KAFKA_HEIGHT,
+  group: false,
+  visible: true,
+  style: {
+    padding: KAFKA_PADDING,
+  },
+};
+
+export const createOverviewItem = (obj: K8sResourceKind): OverviewItem<K8sResourceKind> => {
+  if (!obj.apiVersion) {
+    obj.apiVersion = apiVersionForModel(KafkaConnectionModel);
+  }
+  if (!obj.kind) {
+    obj.kind = KafkaConnectionModel.kind;
+  }
+
+  return {
+    isOperatorBackedService: true,
+    obj,
+  };
+};
+
+export const getTopologyRhoasNodes = (kafkaConnections: K8sResourceKind[]): NodeModel[] => {
+  const nodes = [];
+  for (const obj of kafkaConnections) {
```

Comment:
I think we can make use of the `reduce` func here.

Reply:
I would like to keep it the same as other plugins like helm.